src/components/ClusterNodes/NodeList.tsx

```diff
@@ -0,0 +1,526 @@
+                            if (_clusterId === clusterId) {
+                                setSelectedCluster({
+                                    label: cluster.name,
+                                    value: _clusterId,
+                                })
+                            }
+                            return {
+                                label: cluster.name,
+                                value: _clusterId,
+                            }
+                        })
+                    setClusterList(optionList)
+                }
+            })
+            .catch((error) => {
+                showError(error)
+            })
+    }, [])
+
+    useEffect(() => {
+        const _lastDataSyncTime = Date()
+        setLastDataSyncTimeString('Last refreshed ' + handleUTCTime(_lastDataSyncTime, true))
+        const interval = setInterval(() => {
+            setLastDataSyncTimeString('Last refreshed ' + handleUTCTime(_lastDataSyncTime, true))
+        }, 1000)
+        return () => {
+            clearInterval(interval)
+        }
+    }, [lastDataSync])
+
+    const handleFilterChanges = (): void => {
+        let _flattenNodeList = []
+        for (let index = 0; index < flattenNodeList.length; index++) {
+            const element = flattenNodeList[index]
+            if (selectedVersion.value !== defaultVersion.value && element['k8sVersion'] !== selectedVersion.value) {
+                continue
+            }
+            if (selectedSearchTextType === 'name' && element['name'].indexOf(searchText) === -1) {
+                continue
+            } else if (selectedSearchTextType === 'label') {
+                let matchedLabelCount = 0
+                for (let i = 0; i < element['labels']?.length; i++) {
+                    const currentLabel = element['labels'][i]
+                    const matchedLabel = searchedLabelMap.get(currentLabel.key)
+                    if (matchedLabel === undefined || (matchedLabel !== null && currentLabel.value !== matchedLabel)) {
+                        continue
+                    }
+                    matchedLabelCount++
+                }
+                if (searchedLabelMap.size !== matchedLabelCount) {
+                    continue
+                }
+            }
+            _flattenNodeList.push(element)
+        }
+        if (sortByColumn) {
+            const comparatorMethod =
+                sortByColumn.sortType === 'number' ? numericComparatorMethod : alphabeticalComparatorMethod
+            _flattenNodeList.sort(comparatorMethod)
+        }
+        setFilteredFlattenNodeList(_flattenNodeList)
+        setNoResults(_flattenNodeList.length === 0)
+    }
+
+    const numericComparatorMethod = (a, b) => {
+        let firstValue = a[sortByColumn.sortingFieldName] || 0
+        let secondValue = b[sortByColumn.sortingFieldName] || 0
+        if (typeof firstValue === 'string' && firstValue?.endsWith('%')) {
```

Comment:
no need for an optional check as we've already checked `firstValue` for type string

Reply:
Also, do we need the same check for secondValue as well?

Reply:
Done